UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Layer.cs

```diff
@@ -143,6 +145,13 @@ public virtual void SetTile(Vector3Int position, GenericTile tile, Matrix4x4 tra
 		InternalSetTile(position, tile);
 		tilemap.SetColor(position, color);
 		tilemap.SetTransformMatrix(position, transformMatrix);
+		if (CustomNetworkManager.IsHeadless) return;
+		if (tile is not SimpleTile c) return;
+		if(c.CanBeHighlightedThroughScanners == false || c.HighlightObject == null) return;
+		var spawnHighlight = Spawn.ClientPrefab(c.HighlightObject, MatrixManager.LocalToWorld(position, matrix), this.transform);
+		if(spawnHighlight.Successful == false || spawnHighlight.GameObject.TryGetComponent<HighlightScan>(out var scan) == false) return;
+		c.AssoicatedSpawnedObjects.Add(spawnHighlight.GameObject);
+		scan.Setup(c.sprite);
```

Comment:
Can you put a short comment above? It's a bit too dense to see what it does at a glance.

Reply:
Done